File: app/src/main/java/ch/bailu/aat/map/layer/gpx/GpxLayer.java
package ch.bailu.aat.map.layer.gpx;

import ch.bailu.aat.dispatcher.OnContentUpdatedInterface;
import ch.bailu.aat.gpx.GpxInformation;
import ch.bailu.aat.gpx.GpxList;
import ch.bailu.aat.map.MapContext;
import ch.bailu.aat.map.layer.MapLayerInterface;

public abstract class GpxLayer implements MapLayerInterface, OnContentUpdatedInterface {
    private final int color;

    private GpxList gpxList=GpxList.NULL_ROUTE;


    public GpxLayer(int c) {
        color = c;
    }

    @Override
    public void drawForeground(MapContext mcontext) {}

    @Override
    public void onContentUpdated(int iid, GpxInformation info) {
        setTrack(info.getGpxList());
    }

    private void setTrack(GpxList gpx) {
        if (gpx == null) gpxList = GpxList.NULL_ROUTE;
        else gpxList = gpx;
    }

    public GpxList getGpxList() {
        return gpxList;
    }

    public int getColor() {
        return color;
    }

    @Override
    public void onLayout(boolean changed, int l, int t, int r, int b) {}


}

File: aat-lib/src/main/java/ch/bailu/aat_lib/map/layer/gpx/GpxLayer.kt
package ch.bailu.aat_lib.map.layer.gpx

import ch.bailu.aat_lib.dispatcher.TargetInterface
import ch.bailu.aat_lib.gpx.GpxInformation
import ch.bailu.aat_lib.gpx.GpxList
import ch.bailu.aat_lib.map.MapColor.getColorFromIID
import ch.bailu.aat_lib.map.MapContext
import ch.bailu.aat_lib.map.layer.MapLayerInterface
import ch.bailu.aat_lib.preferences.StorageInterface
import ch.bailu.aat_lib.util.Point

abstract class GpxLayer : MapLayerInterface, TargetInterface {
    var colorFromIID = 0
        private set

    var gpxList = GpxList.NULL_ROUTE
        private set

    override fun drawForeground(mcontext: MapContext) {}
    override fun onContentUpdated(iid: Int, info: GpxInformation) {
        gpxList = info.getGpxList() ?: GpxList.NULL_ROUTE
        colorFromIID = getColorFromIID(iid)
    }

    override fun onLayout(changed: Boolean, l: Int, t: Int, r: Int, b: Int) {}

    override fun onTap(tapPos: Point): Boolean {
        return false
    }

    override fun onAttached() {}
    override fun onDetached() {}
    override fun onPreferencesChanged(storage: StorageInterface, key: String) {}
}
